### DISCUSSION TITLE
Reasoning for control model

### DISCUSSION THREAD

**Discussion by braggbear** [sibylhe/mmm_stan#25] [Q&A]
Hi Sibyl! What is the reasoning behind doing the control model separate from the media? Why not do them all together in one model?  Also, if you don't account for media spend in the control model, won't you get incorrect coefficients? Shouldn't the control model know how much media impact there was so it doesn't incorrectly give credit to the control variables? Thanks!

**Comment by sibylhe** [comment]
Definitely the control model and media model can be combined into one hierarchical Bayesian model, that's the ideal way.

Separating them can ease convergence. It's a pain point for me because I have many parameters to estimate (46 control coefficients + 13 media coefficients + 26 adstock parameters), and some media variables are correlated. If your sales is largely determined by non-marketing (control) effect, you can separate it into two models. This model is done for retail company. The market is fully competitive, price match is only one click away. The majority of sales is determined by price, holiday, macroeconomy - usually these non-marketing factors account for 70-80% of sales, marketing 20-30%. The control model (sales ~ control variables) aims to predict the non-marketing sales trend, the actual sales roughly represent the non-marketing sales trend since 80% of sales is determined by non-marketing factors. So it's okay to use the actual sales as target variable here. I expect this separate-and-stack way could yield similar result as hierarchical model (Haven't tried a hierarchical model, if you decide to give it a try, please do let me know how long it takes to converge/whether it has a good convergence. I'm very interested).

It depends on your company and industry. If your sales relies heavily on marketing (e.g., 20% non-marketing + 80% marketing), go for the hierarchical model.

**Reply by maswiebe** [reply]
Hi Sibyl, why didn't you residualize the control variables, like [Jin et al.](https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/46001.pdf) do:

> The control variables are highly correlated, as shown in Figure 13a, which induces high correlation
in the posteriors of the regression coefficients γ c , c = 1, ... , 3. Including them directly in the model
would cause the MCMC sampler to take a much longer time to converge. Instead, we make them
nearly orthogonal by regressing each of distribution and promotion onto price and use the residuals
of the single variate linear regression as a predictor in the model.
